controllers/authController.js

```diff
@@ -0,0 +1,146 @@
+const { promisify } = require("util");
+const jwt = require("jsonwebtoken");
+const User = require("../models/userModel");
+const AppError = require("../utils/appError");
+
+const createToken = id => {
+    return jwt.sign(
+        {
+            id,
+        },
+        process.env.JWT_SECRET,
+        {
+            expiresIn: process.env.JWT_EXPIRES_IN,
```

Comment:
algorithm: ? (ECDSA, HMAC ?)
Do we use JWT for encryption or signing ?

Reply:
https://jwt.io/introduction/
Digital signature. We can choose to use 

   - HMAC + SHA256
   - RSASSA-PKCS1-v1_5 + SHA256
   - ECDSA + P-256 + SHA256
   
   Normally, the defaul is HMAC

Reply:
Do we add "role" in the payload too ?

Reply:
Will do